### PR TITLE
kernel/binary_manager : Recover mq and sem before state changes

### PR DESCRIPTION
**Pull request by YiHyunJin**
…uling

We need to recover mq and sem before state changes.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [d54b0a44517a3a419c625d55156db5bba0f2e1e9] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [d54b0a44517a3a419c625d55156db5bba0f2e1e9] - Code Rule Check (C++) OK.

File: os/kernel/binary_manager/binary_manager_recovery.c [hunk 1]
```diff
@@ -74,6 +76,8 @@ static void recovery_exclude_scheduling_each(FAR struct tcb_s *tcb, FAR void *ar
 	if (tcb->group->tg_loadtask == binid) {
 		flags = irqsave();
 		/* Remove the TCB from the task list associated with the state */
+		sem_recover(tcb);
```
Comment (jeongarmy) [inline os/kernel/binary_manager/binary_manager_recovery.c]: please add comment about this.

**Comment by seinfra** [comment]
Target : [f527c59f8ff243480fb3eef14b5f387d96139cec] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [f527c59f8ff243480fb3eef14b5f387d96139cec] - Code Rule Check Result:
os/kernel/binary_manager/binary_manager_recovery.c:88: ERROR: [BRC_M_SMT] that open brace { should be on the previous line
os/kernel/binary_manager/binary_manager_recovery.c:122: ERROR: [SPC_M_OPR] space prohibited before that ':' (ctx:WxE)
os/kernel/binary_manager/binary_manager_recovery.c:127: ERROR: [SPC_M_OPR] space prohibited before that ':' (ctx:WxE)

**Comment by seinfra** [comment]
Target : [7b0cc5b724d27618888935cf2c769d6130a978ae] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [7b0cc5b724d27618888935cf2c769d6130a978ae] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [e5940b3733323f76712ae62e8529dde3cc5a44d8] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [e5940b3733323f76712ae62e8529dde3cc5a44d8] - Code Rule Check (C++) OK.

**Pull request merged**